src/Qir/Runtime/stdlib/src/callables.rs

```diff
@@ -0,0 +1,171 @@
+                // list created when performing concatenate above. In the latter case, the concatenated
+                // list will get cleaned up, preventing memory from leaking.
+                __quantum__rt__array_update_reference_count(
+                    *args_copy.cast::<*const QirArray>(),
+                    -1,
+                );
+                // Decrement the count on the copy to clean it up as well, since we created a new copy
+                // with the updated controls list.
+                __quantum__rt__tuple_update_reference_count(args_copy, -1);
+                args_copy = new_args;
+                ctl_count -= 1;
+            }
+        }
+    }
+
+    (*call
+        .func_table
+        .wrapping_add(index)
+        .cast::<extern "C" fn(*mut u8, *mut u8, *mut u8)>())(
+        call.cap_tuple,
+        args_copy.cast::<u8>(),
+        res_tup,
+    );
+    if *call.ctls_count.borrow() > 0 {
+        __quantum__rt__array_update_reference_count(*args_copy.cast::<*const QirArray>(), -1);
+    }
+    if !args_copy.is_null() {
+        __quantum__rt__tuple_update_reference_count(args_copy, -1);
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn __quantum__rt__callable_copy(
+    callable: *const Callable,
+    force: bool,
+) -> *const Callable {
+    let rc = ManuallyDrop::new(Rc::from_raw(callable));
+    if force || Rc::weak_count(&rc) > 0 {
+        let copy = rc.as_ref().clone();
+        Rc::into_raw(Rc::new(copy))
+    } else {
+        Rc::into_raw(Rc::clone(&rc));
+        callable
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn __quantum__rt__callable_make_adjoint(callable: *const Callable) {
+    let call = &*callable;
+    call.is_adj.replace_with(|&mut old| !old);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn __quantum__rt__callable_make_controlled(callable: *const Callable) {
+    let call = &*callable;
+    call.ctls_count.replace_with(|&mut old| old + 1);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn __quantum__rt__callable_update_reference_count(
+    callable: *const Callable,
+    update: i32,
+) {
+    update_counts(callable, update, false);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn __quantum__rt__callable_update_alias_count(
+    callable: *const Callable,
+    update: i32,
+) {
+    update_counts(callable, update, true);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn __quantum__rt__capture_update_reference_count(
+    callable: *const Callable,
+    update: i32,
+) {
+    let call = &*callable;
+    if !call.mem_table.is_null() && !(*(call.mem_table)).is_null() {
+        (*call.mem_table.cast::<extern "C" fn(*mut u8, i32)>())(call.cap_tuple, update);
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn __quantum__rt__capture_update_alias_count(
+    callable: *const Callable,
+    update: i32,
+) {
+    let call = &*callable;
+    if !call.mem_table.is_null() && !(*(call.mem_table.wrapping_add(1))).is_null() {
+        let _val = **(call.mem_table.cast::<*mut usize>().wrapping_add(1));
```

Comment:
What does this line do?

Reply:
Oh, great catch! This is leftover from debugging investigation in an earlier prototype and was accidentally copied into here. It should be removed.